### PR TITLE
Release v0.17.0

### PR DESCRIPTION
**Pull request by TobiGr**
#### Playback resume #2288
- Resume streams where you stopped last time
#### Downloader Enhancements #2149 
- Use Storage Access Framework to store downloads on external SD-cards
- New mp4 muxer
- Optionally change the download directory before starting a download
- Respect metered networks 

---------

### Improved
- Removed gema strings #2295
- Handle (auto)rotation changes during activity lifecycle #2444
- Make long-press menus consistent #2368

---------

### Fixed
- Fixed selected subtitle track name not being shown #2394
- Do not crash when check for app update fails (GitHub version) #2423 
- Fixed downloads stuck at 99.9% #2440
- Update play queue metadata #2453 
- [SoundCloud] Fixed crash when loading playlists TeamNewPipe/NewPipeExtractor#170
- [YouTube] Fixed duration can not be paresd TeamNewPipe/NewPipeExtractor#177
- [YouTube] Fixed crash on videos which are age restricted TeamNewPipe/NewPipeExtractor#177

---------

### Blockers

- [x] Slow download speed of audio streams

- [x] Database import fails
- [ ] History loading is slow

---------

Corresponding issue: #2483 
Corresponding PR for the extractor: https://github.com/TeamNewPipe/NewPipeExtractor/pull/179

### APKs for testing and finding bugs
are available in the corresponding issue:

**Please report all new bugs which are related to one of the changes above in #2483**

### CODE REVIEW

**Comment by TobiGr** [comment]
I've been using the new version for quite some time now and couldn't find any blocking bugs. @theScrabi is going to upload an offically signed apk soon. As I didn't encountered any new bugs and hopefully you won't either, I'd like to release 0.17.0 after a short testing period at the end of this week or at the beginning of the next week.

**Comment by theScrabi** [comment]
I'd also open an issue for this, as I think people read more issues than pull requests.

File: app/build.gradle [hunk 1]
```diff
@@ -57,7 +57,7 @@ dependencies {
         exclude module: 'support-annotations'
     })
 
-    implementation 'com.github.TeamNewPipe:NewPipeExtractor:5f65788a2f89e'
+    implementation 'com.github.TeamNewPipe:NewPipeExtractor:4488c217659d2e92'
```
Comment (TobiGr) [inline app/build.gradle]: @theScrabi What about https://github.com/TeamNewPipe/NewPipeExtractor/pull/177? I've already updated to https://github.com/TeamNewPipe/NewPipeExtractor/commit/5f65788a2f89eedbf384aa5c5f3f98dcfd73400e but you downgrade to https://github.com/TeamNewPipe/NewPipeExtractor/commit/4488c217659d2e923f125fd923e1b28812cf217f.

**Comment by TobiGr** [comment]
> I'd also open an issue for this, as I think people read more issues than pull requests.

I wanted to do this as soon as there are apks for testing.

File: app/build.gradle (hunk 1 above)
Comment (theScrabi) [reply app/build.gradle]: ups ... forgot to pull the extractor sorry

**Comment by theScrabi** [comment]
I quickly removed that last commit and rebuild the app.

fast

**Comment by TobiGr** [comment]
>Or does WebLate do this automatically as soon as this pr is merged? 

@Stypox Yes, but I don't wan't to have these unused strings in the release APK. I removed the left over gema strings during the merge, but forgot `direct_on_background` . I'll remove them later today.

**Comment by snappyapple632** [comment]
Hmm... If MP4 support is going to be added, I think it would be also a good time to address #507 in a future update, since MP4 is essentially MP3 + video. I'm sure the MP4 muxer could be modified to also serve as a MP3 muxer.

**Pull request merged**